### PR TITLE
Project Survey - Sara Heijkenskjöld

### PR DESCRIPTION
**Pull request by sarahejhej**
Link to my deployed project:
https://excellent.netlify.app/

My reflections on how this week's project turned out:
I think it went quite well this week. It was fun trying out Sass, but I'm not sure I managed to combine it with BEM. The class naming felt a bit messy when different code blocks overlapped. 

Things I'd like to have clarified or explained in more detail:
I would like some tips on how to clean up the Sass and BEM and how to disable the proceed button in the checkbox page in the form.

### CODE REVIEW

File: code/src/styles/Form.scss [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    padding: 0 0 0 30px;
+    margin: 10px 5px;
+    cursor: pointer;
+    font-size: 16px;
+    -webkit-user-select: none;
+    -moz-user-select: none;
+    -ms-user-select: none;
+    user-select: none;
+
+    & input {
```
Comment (Andrea-Osmar) [inline code/src/styles/Form.scss:64]: I think it is to much indentation here?

File: code/src/styles/Form.scss [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    height: 43px;
+    font-size: 22px;
+    color: rgba(247, 212, 184, 0.8);
+
+    &::placeholder {
+      font-size: 20px;
+    }
+  }
+
+  /*--- custom radio and checkbox input ---*/
```
Comment (Andrea-Osmar) [inline code/src/styles/Form.scss:50]: love the comments to make it easier to follow :)

File: code/src/variables.scss [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  @if ($breakpoint == medium) {
+    @media only screen and (min-width: 768px) {
+      @content;
+    }
+  }
+  @if ($breakpoint == large) {
+    @media only screen and (min-width: 1024px) {
+      @content;
+    }
+  }
```
Comment (Andrea-Osmar) [inline code/src/variables.scss:11]: the survey looks really good in ipad & iphone view!

**Review by Andrea-Osmar — APPROVED** [review]
Very good job Sara!
I really like your design and layout, its very clean. I also enjoy reading your code since its very structured and easy to follow. However I am not familiar with Sass at all so I had a bit of a hard time to understand, neet to ready up on Sass!

I tabed my way thru your site and it worked well. However when I tried to use David for screenreading I didnt get him to read the questions.
I ran your site thru Lighthouse as well and got a perfect result back! 

Your servey fulfill all the general and blue requirements as well as most of the red and black requirements! 
Really impressive! 

I learned a lot from reading your code and I got very interested in learning Sass! You structure your code really well and it is easy to follow your functions.

I don't have much feedback, just keep up the good work! And maybe put a little more time on the questions to keep them a part a bit ;)

File: code/src/components/App.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import React from 'react';
+import Form from './Form';
```
Comment (EmOlsz) [inline code/src/components/App.js:2]: Next time one line of empty space between outer and local dependencies would be great :)

File: code/src/components/Form.js [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+    setPageIndex(pageIndex + 1);
+  };
+  const onPreviousPageChange = () => {
+    setPageIndex(pageIndex - 1);
+  };
+  const onRefresh = () => {
+    window.location.reload();
+  };
+
+  const selectArray = [
```
Comment (EmOlsz) [inline code/src/components/Form.js:49]: I'm very happy to see such clean approach with arrays. There is one thing we could improve even more - create separate file, something like `utils.js` or `data.js` and store all arrays and objects there

File: code/src/components/Form.js [hunk 6]
```diff
@@ -0,0 +1,305 @@
...
+
+  return (
+    <main className="App__main">
+      {!submit ? (
+        <form className="Form" onSubmit={handleSubmit}>
+          <h1 className="App__header">Excellent survey</h1>
+
+          {pageIndex === -1 && (
+            <section className="Form__section">
+              <h2 className="App__subheader">
```
Comment (EmOlsz) [inline code/src/components/Form.js:96]: Appreciate BEM - in your case as always - but we should stick to lowercase naming :

```
<section className="form__section">
    <h2 className="app__subheader">
```

File: code/src/components/Form.js [hunk 7]
```diff
@@ -0,0 +1,305 @@
...
+                <Button className="Button" type="submit" text="Submit" />
+              </div>
+            </section>
+          )}
+          {pageIndex >= 0 && (
+            <section className="Form__section Form__section-progressbar">
+              <ProgressBar
+                classLabel="Form__label-progressbar"
+                progress={pageIndex}
+                maxProgress={Object.keys(questions).length}
```
Comment (EmOlsz) [inline code/src/components/Form.js:284]: Great!!!

File: code/src/components/InputRadiobutton.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+  array,
+  value,
+  onRadioChange,
+  classSpan,
+}) => {
+  return (
+    <>
+      <h2>{question}</h2>
+      {array.map(radio => (
+        <label className={classLabel} htmlFor={radio} key={radio} tabIndex="0">
```
Comment (EmOlsz) [inline code/src/components/InputRadiobutton.js:15]: No need for `tabindex` in `label`, they have this functionality built in

File: code/src/styles/App.scss [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  background: #f7d4b8;
+  height: 100vh;
+  font-family: 'Montserrat', sans-serif;
+  color: rgb(247, 212, 184);
+
+  &__main {
```
Comment (EmOlsz) [inline code/src/styles/App.scss:13]: You seem to impress me with every next project Sara. Great usage of ampersand, this is exactly how we should combine BEM with Sass features!

**Review by EmOlsz — COMMENTED** [review]
Title of your survey tells everything - pure excellence, nothing else to add :)

**Pull request closed without merging**